scrabble.js

```diff
@@ -1,16 +1,159 @@
 
+const scrabbleTiles = {
+  a: 1,
+  e: 1,
+  i: 1,
+  o: 1,
+  u: 1,
+  l: 1,
+  n: 1,
+  r: 1,
+  s: 1,
+  t: 1,
+  d: 2,
+  g: 2,
+  b: 3,
+  c: 3,
+  m: 3,
+  p: 3,
+  f: 4,
+  h: 4,
+  v: 4,
+  w: 4,
+  y: 4,
+  k: 5,
+  j: 8,
+  x: 8,
+  q: 10,
+  z: 10,
+};
+
 const Scrabble = {
+  isLetter(letter) {
+    return /[a-z]/.test(letter);
+  },
+
   score(word) {
+    word = word.toLowerCase();
+    let totalScore = 0;
+
+    if (word.length > 7) {
+      throw 'Invalid word: too many tiles.';
+    } else if (word.length === 0) {
+      throw 'Invalid word: must provide tiles.';
+    } else if (word.length === 7) {
+      totalScore += 50;
+    }
+
+    for (let letter of word) {
+      if (this.isLetter(letter)) {
+        totalScore += scrabbleTiles[letter];
+      } else {
+        throw 'Invalid word: must include letters that are acceptable.';
+      }
+    }
 
+    return totalScore;
   },
+
   highestScoreFrom(arrayOfWords) {
+    let topWord = arrayOfWords[0];
+
+    if (typeof arrayOfWords !== 'object' || arrayOfWords.length === 0) {
+      throw 'Invalid array.';
+    }
+
+    arrayOfWords.forEach((word) => {
+      if (this.score(word) > this.score(topWord)) {
+        topWord = word;
+      } else if (this.score(word) === this.score(topWord)) {
```

Comment:
While this totally works, it is not the most efficient solution. You are calling the `score` method for the same word multiple times. I'd recommend storing this result in a variable before executing these conditionals